### PR TITLE
feat(apps-landing): add community email modal

### LINKED ISSUE #1994: Landing Page Modifications
If you need to add env to vercel, do it in terraform. I'll add some docs on how to achieve that. Don't add directly to Vercel

- [x] Mailchimp

On the landing page, the `Join The Community` should bring up a modal with email address field, this will integrate with Mailchimp to collect email address.

After success close modal and show an antd success message component on the top right

- [x] Intercom

Integrate intercom chat for communicating with new potential customers on our landing page.

I've sent invite for this

- [x] Google analytics
- [x] Hotjar for heatmap tracking
- [ ] SEO optimization

We'll need to start SEO, go through this list and add anything we're missing https://mailchimp.com/resources/website-search-engine-optimization-checklist/. Add to a list here for each item you add.

- Title tag, meta description (use next-seo)
-  Alt tag for images

- [ ] Speed optimization

Make sure build is minified
https://pagespeed.web.dev/ speed test should be good, screenshot results here

- [ ] Integrate Supabase to resource page

Create a `Resource` model with `title`, `cover`, and `description` to link some data to our resource page. Add some demo data and display a list of resources on our resource page

- [ ] Meta pixel 

Send me your facebook profile, since I need to add you as friend in order to add you as a user to the codelab facebook page, where you can setup the pixel configurations

- [x] Refactor `shared/env`

Split `env.ts` into two `landing-env.ts` and `builder-env.ts`. in case there are shared elements, create a third one `shared-env.ts`

### PR DESCRIPTION
**Pull request by JLL32**
## Description (Optional)

Add a modal for the community feature to collect audience emails
### States
modal
Screenshot 2022-11-19 at 00 51 51

modal, valid email
Screenshot 2022-11-19 at 00 52 14

modal, invalid email
Screenshot 2022-11-19 at 00 52 24

notification, success
Screenshot 2022-11-19 at 00 53 42

notification, failure
Screenshot 2022-11-19 at 00 53 22

## Related Issue(s)

Fixes #1994

### CODE REVIEW

File: apps/landing/home/community/EmailModal.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
+import { Button, Input, Modal } from 'antd'
+import React, { useState } from 'react'
+import tw from 'twin.macro'
+import { z } from 'zod'
+
+interface EmailModalProps {
```
Comment (webberwang) [inline apps/landing/home/community/EmailModal.tsx:6]: why recreate interface when we can re-use existing?

File: apps/landing/pages/api/community/email.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import client from '@mailchimp/mailchimp_marketing'
+import { NextApiRequest, NextApiResponse } from 'next'
+import z from 'zod'
+
+const LIST_ID = process.env.MAILCHIMP_LIST_ID || ''
```
Comment (webberwang) [inline apps/landing/pages/api/community/email.ts:5]: possible to move to `shared-env`?

let's have `landing-env.ts` and `builder-env.ts`

**Review by webberwang — CHANGES_REQUESTED** [review]

File: apps/landing/home/community/EmailModal.tsx (hunk 1 above)
Comment (JLL32) [reply apps/landing/home/community/EmailModal.tsx]: The defined type is a bit different, I have expressed that difference in the new declaration

File: apps/landing/pages/api/community/email.ts (hunk 2 above)
Comment (JLL32) [reply apps/landing/pages/api/community/email.ts]: addressed

**Pull request merged**